lib/tbot/workloadidentity/workloadattest/podman.go

```diff
@@ -0,0 +1,143 @@
+	// Addr is the address at which the [Podman API service] can be reached.
+	//
+	// It must be in the form `unix://path/to/socket`, as making the service
+	// reachable via the network (i.e. binding to a TCP port) is *strongly*
+	// discouraged for security reasons.
+	//
+	// Addr is a required option. We do not provide a default value because
+	// there isn't one that would work "out of the box" in the majority of cases.
+	//
+	// # Rootful Podman
+	//
+	// If running Podman as root, enable and start the API service by running:
+	//
+	// 	$ sudo systemctl enable --now podman.socket
+	//
+	// At this point, the service can be reached at `unix:///run/podman/podman.sock`
+	// but only by root. If you do not want to run tbot as root, you can override
+	// the default systemd unit to make the socket owned by a group your tbot user
+	// belongs to.
+	//
+	// 	$ sudo groupadd podman-root
+	// 	$ sudo usermod -a -G podman-root <tbot user>
+	// 	$ sudo systemctl edit podman.socket
+	//
+	// In the override file add:
+	//
+	// 	[Socket]
+	// 	SocketGroup=podman-root
+	//
+	// Remember to reload the systemd daemon and restart the socket unit:
+	//
+	// 	$ sudo systemctl daemon-reload
+	// 	$ sudo systemctl restart podman.socket
+	//
+	// # Rootless Podman
+	//
+	// If running Podman as a non-root user, enable and start the API service by
+	// running:
+	//
+	// 	$ sudo loginctl enable-linger <podman user>
+	// 	$ systemctl --user enable --now podman.socket
+	//
+	// The service can now be reached at `unix://$XDG_RUNTIME_DIR/podman/podman.sock`
+	// (i.e. `/run/user/$(id -u)/podman/podman.socket`) but only by your chosen
+	// Podman user or root. If you do not want to run tbot as the same user or
+	// as root, you can override the default systemd unit to move the socket out
+	// of `$XDG_RUNTIME_DIR` and add your tbot user to the podman user's group.
+	//
+	// 	$ sudo usermod -a -G <podman user group> <tbot user>
+	// 	$ systemctl --user edit podman.socket
+	//
+	// In the override file add:
+	//
+	// 	[Socket]
+	// 	ListenStream=/path/to/socket # Example: /tmp/podman.<podman user>.sock
+	//
+	// Remember to reload the systemd daemon and restart the socket unit:
+	//
+	// 	$ systemctl --user daemon-reload
+	// 	$ systemctl --user restart podman.socket
+	//
+	// # Running tbot in a container
+	//
+	// In order for tbot running inside a container to attest workloads in other
+	// containers, it must have access to the host's PID namespace. Also, if you
+	// are using group permissions to grant access to the Podman socket (see the
+	// above examples) you must set the `run.oci.keep_original_groups` annotation.
+	//
+	// 	$ podman run \
+	// 		--pid=host \
+	// 		--annotation run.oci.keep_original_groups=1 \
+	// 		--volume /path/to/socket:/path/to/socket \
+	// 		--volume /path/to/config:/path/to/config \
+	// 		<tbot image> \
+	// 		start -c /path/to/config
+	//
+	// [Podman API service]: https://docs.podman.io/en/latest/markdown/podman-system-service.1.html
```

Comment:
☝🏻 Here's the thought behind the `addr` field's lack of default I mentioned.

Reply:
It's just occurred to me that we probably shouldn't be recommending folks put their socket into `/tmp` because of the risk of [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) vulnerabilities. I'll update that example to say `/srv/podman.<podman user>/podman.sock` instead, although that is slightly more hassle to set up.

Reply:
It's just occurred to me that we probably shouldn't be recommending folks put their socket into `/tmp` because of the risk of [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) vulnerabilities. I'll update that example to say `/srv/podman.<podman user>/podman.sock` instead, although that is slightly more hassle to set up.